[timeseries] Add custom forecasting model tutorial

shchur opened the pull request:
*Issue #, if available:*

*Description of changes:*
- Add a tutorial describing how to add a custom time series forecasting model


By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

abdulfatir reviewed (approved):
Read through the notebook. Everything looks good to me, very comprehensive. I am just wondering if the "Using the custom model in standalone mode" section is strictly required.

shchur commented:
> Read through the notebook. Everything looks good to me, very comprehensive. I am just wondering if the "Using the custom model in standalone mode" section is strictly required.

Thanks @abdulfatir, I've added a disclaimer that "Standalone mode" should only be used for development and debugging and also fixed some layout issues.
